### PR TITLE
[improvement](spill) improve config and fix spill bugs

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run feut

**Comment by jacktengg** [comment]
run buildall

File: be/src/common/config.cpp [hunk 1]
```diff
@@ -1160,13 +1160,13 @@ DEFINE_mDouble(high_disk_avail_level_diff_usages, "0.15");
 DEFINE_Int32(partition_disk_index_lru_size, "10000");
 // limit the storage space that query spill files can use
 DEFINE_String(spill_storage_root_path, "${DORIS_HOME}/storage");
-DEFINE_mInt64(spill_storage_limit, "10737418240"); // 10G
-DEFINE_mInt32(spill_gc_interval_ms, "2000");       // 2s
+DEFINE_mInt32(spill_storage_usage_percent, "20"); // 20%
```
Comment (yiguolei) [inline be/src/common/config.cpp:1163]: Do not write config like this.
Maybe there are 2 requirements:
1. Set the usage by percent like you do.
2. Set the usage by size, such as 10g, 10T, 100mb

If you add a config like this, then you has to add another config spill_storage_usage_bytes when meet req 2.
If user set both of them, we do not know which should be used.

You could refer the config mem_limit.

File: be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp [hunk 2]
```diff
@@ -659,12 +662,13 @@ Status PartitionedHashJoinProbeOperatorX::pull(doris::RuntimeState* state,
...
         local_state._need_to_setup_internal_operators = false;
+        auto& mutable_block = local_state._partitioned_blocks[partition_index];
+        if (mutable_block && !mutable_block->empty()) {
+            probe_blocks.emplace_back(mutable_block->to_block());
+        }
     }
-
-    auto partition_index = local_state._partition_cursor;
-    bool in_mem_eos_;
+    bool in_mem_eos_ = false;
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp:670]: 普通的变量不要以 _ 结尾了

File: be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp [hunk 3]
```diff
@@ -682,7 +686,9 @@ Status PartitionedHashJoinProbeOperatorX::pull(doris::RuntimeState* state,
 
         auto block = std::move(probe_blocks.back());
         probe_blocks.pop_back();
-        RETURN_IF_ERROR(_inner_probe_operator->push(runtime_state, &block, false));
+        if (!block.empty()) {
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp:689]: 这里如果block 是empty，然后也不忘probe operator 里push ，那么672 行会不会死循环？

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Review by mrhhsg — DISMISSED** [review]
LGTM

**Comment by jacktengg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by mrhhsg — APPROVED** [review]
lgtm

**Pull request merged**